resources/js/Components/Comment.tsx

```diff
@@ -0,0 +1,140 @@
+import React, { useState } from 'react';
+import axios from 'axios';
+import classNames from 'classnames';
+import { usePage } from '@inertiajs/react';
+
+import { CommentType, PageProps, PostType, User } from '@/types';
+import { formatDate } from '@/utils';
+import CommentBox from './CommentBox';
+
+type CommentProps = {
+  comment: CommentType;
+  post: PostType;
+  parentId?: number;
+  onCommentDelete: (commentId: number) => void;
+};
+
+const Comment = ({
+  post,
+  comment,
+  parentId,
+  onCommentDelete,
+}: CommentProps) => {
+  const [showReplyBox, setShowReplyBox] = useState(false);
+  const { auth } = usePage<PageProps>().props;
+  const [commentParentId, setCommentParentId] = useState(parentId);
```

Comment:
You can avoid this state and prevent re-renders. This case just use the parentId directly. It's never changed for a this components.

Reply:
You can find the improvement here, https://gist.github.com/alaminfirdows/f0e40dda5b33150640824675f7a383ea